atlasdb-config/src/test/java/com/palantir/atlasdb/http/AtlasDbHttpClientsTest.java

```diff
@@ -172,6 +175,41 @@ public void canLiveReloadServersList() {
         unavailableServer.verify(getRequestedFor(urlMatching(TEST_ENDPOINT)));
     }
 
+    @Test
+    public void httpProxyThrowsServiceNotAvailableExceptionIfConfiguredWithZeroNodes() {
+        TestResource testResource = AtlasDbHttpClients.createLiveReloadingProxyWithQuickFailoverForTesting(
+                () -> ImmutableServerListConfig.builder().build(),
+                SslSocketFactories::createSslSocketFactory,
+                proxyConfiguration -> ProxySelector.getDefault(),
+                TestResource.class,
+                UserAgents.DEFAULT_VALUE);
+
+        assertThatThrownBy(testResource::getTestNumber).isInstanceOf(ServiceNotAvailableException.class);
+    }
+
+    @Test
+    public void httpProxyCanBeCommissionedAndDecommissionedIfNodesBecomeUnavailable() {
+        AtomicReference<ServerListConfig> config = new AtomicReference<>(ImmutableServerListConfig.builder().build());
+
+        TestResource testResource = AtlasDbHttpClients.createLiveReloadingProxyWithQuickFailoverForTesting(
+                config::get,
+                SslSocketFactories::createSslSocketFactory,
+                proxyConfiguration -> ProxySelector.getDefault(),
+                TestResource.class,
+                UserAgents.DEFAULT_VALUE);
+        assertThatThrownBy(testResource::getTestNumber).isInstanceOf(ServiceNotAvailableException.class);
```

Comment:
This needs a bit of clarification - looks like we use defaults; why should it be unavailable?
(Because there are no nodes yet, yes, but someone reading only this test might not remember that)

Reply:
Yes. The line uses defaults, apart from the `config::get` (which is initially empty). Added a comment!